dash/backend/src/modules/falco/controllers/falco.controller.ts

```diff
@@ -88,27 +93,51 @@ export class FalcoController {
 
 
     @Post(':clusterid/create')
-    // @TODO: Update to use same authorization method as kube-bench and kube-hunter
     async createFalcoLog(
         @Param('clusterid') clusterId: number,
         @Body() falcoLog: FalcoWebhookInputDto,
         @Query('key') key: string
-    ): Promise<number> {
-
-        //find user authority by user's apikey
+    ): Promise<FalcoDto> {
+        // look up user by api key
         const currentUserAuthObj = await this.userDao.loadUserByApiKey(key);
-        // if the current user's api is valid
-        if (currentUserAuthObj !== null && currentUserAuthObj != undefined ) {
-            // get all authorities from the current user
-            const currentUserAuth = currentUserAuthObj[0].authorities;
-            // get Falco authority from Authorityid
-            let authorityArr: AuthorityId [] = [AuthorityId.FALCO];
-            // is the user a Falco user
-            let isFalcoUser = this.authService.checkAuthority(currentUserAuth, authorityArr);
-            if (isFalcoUser) {
-                return this.falcoService.createFalcoLog(clusterId, falcoLog);
-            }
+        if (!currentUserAuthObj) {
+            return; // @TODO: Should throw access denied exception
         }
+
+        // get all authorities from the current user
+        const currentUserAuth = currentUserAuthObj[0].authorities;
+        let isFalcoUser = this.authService.checkAuthority(currentUserAuth, [AuthorityId.FALCO]);
+        if (!isFalcoUser) {
+            return; // @TODO: Should throw access denied exception
+        }
+
+        // check if falco settings rules should be applied
+        /*const rules = getFalcoSettingsRules(clusterId);
+        for (rules : rule) {
+            if (rule.isRelevant(event)) {
+                if (rule.affect === 'ignore') {
+                    return; // don't save this event - it is being filtered out
+                }
+            }
+        }*/
+
+        // Saved new falco log
+        const newFalcoLog = await this.falcoService.createFalcoLog(clusterId, falcoLog);
+
+        // look up falco settings in order to decide if we need to send email(s) to administrators
+        await this.falcoService.sendFalcoEmailAlert(clusterId, newFalcoLog);
+    }
+
+
+    @Post(':clusterid/settings')
+    async createFalcoSetting(
+        @Param('clusterid') clusterId: number,
+        @Body() falcoSetting: FalcoSettingDto
+    ): Promise <any> {
+        console.log('backend falco controller');
+        console.log('falcosetting:', falcoSetting);
+        const result = this.falcoService.createFalcoSetting(clusterId, falcoSetting);
```

Comment:
After this line it should call this.sendFalcoEmailAlert() so that whenever a new falco event is created (from anywhere in the application) it will always send an email if it is appropriate to do so.

Reply:
This is the createFalcoSettings. Was your comment for createFalcoLog? I think we already have that covered.

Reply:
Oh yeah, true